### PR TITLE
replace rusoto_s3: part 2 (last part)

### PR DESCRIPTION
**Pull request by jychen7**
### Why

step 2 of https://github.com/roapi/roapi/issues/240: replace rusoto_s3 (in use_memory_table: true path) with S3 client from object_store (s3://)

rusoto_s3 is used in `columnq::io::s3` and `columnq::io::s3` is used in two places:
- `columnq::partitions_from_table_source`
- `columnq::table::delta`

we would like to remove rusoto_s3 and switch to object_store, so can unify S3/GCS/Azure in future

### How

following part 1 (https://github.com/roapi/roapi/pull/244), this PR (part 2) will focus on migrating `columnq::table::delta` and delete `columnq::io::s3`

this PR also remove crate `rusoto_xxx` completely and close https://github.com/roapi/roapi/issues/243

### Test

Rely on existing unit tests in 
https://github.com/roapi/roapi/blob/1e3644ec36e1a9080dac3d5f50e3f24990f4cae7/columnq/src/table/delta.rs#L131-L132
https://github.com/roapi/roapi/blob/1e3644ec36e1a9080dac3d5f50e3f24990f4cae7/columnq/src/table/delta.rs#L151-L152

### CODE REVIEW

File: columnq/Cargo.toml [hunk 1]
```diff
@@ -41,11 +41,6 @@ calamine = "0.19.1"
 
 tokio = { version = "1", features = ["rt-multi-thread"] }
 futures = "0.3"
-# S3
```
Comment (jychen7) [inline columnq/Cargo.toml:44]: now s3 is refactor, let me create a release PR for `0.8.2` after this PR is approved and merged.

after we adds GCS and Azure support, I think we can release `0.9.0`?

Comment (houqp) [reply columnq/Cargo.toml:44]: that sounds good to me, let's do it!

**Review by houqp — APPROVED** [review]

**Comment by houqp** [comment]
One thing I forgot to mention in the previous PR review, we should enable the aws-profile feature in the object store crate because using profile to inject credential is a very common pattern in AWS land.

**Comment by jychen7** [comment]
> One thing I forgot to mention in the previous PR review, we should enable the aws-profile feature in the object store crate because using profile to inject credential is a very common pattern in AWS land.

good point, improve here: https://github.com/roapi/roapi/pull/247

**Pull request merged**